src/api/Feed.js

```diff
@@ -517,26 +518,45 @@ class Feed extends Base {
         this.errorCallback = errorCallback;
         this.tasksNewAPI = new TasksNewAPI(this.options);
         this.updateFeedItem({ isPending: true }, task.id);
-        this.tasksNewAPI.updateTask({
-            file,
-            task,
-            successCallback: (taskData: Task) => {
-                this.updateFeedItem(
-                    {
-                        ...taskData,
-                        isPending: false,
-                    },
-                    task.id,
+
+        Promise.resolve()
+            .then(() => {
+                return Promise.all(
```

Comment:
I think we may be able to avoid nesting these `Promise.all` calls. The subsequent calls can be chained off this first `Promise.all`. I think it may help to move each of these to their own methods, as well, with each accepting and resolving the parent `task` object.

```
return this.addAssignees(task)
    .then(this.removeAssignees)
    .then(this.updateTask)
    .catch(() => // catch all the things)
```

Reply:
also maybe more async/await syntax?

Reply:
I could remove the outer `Promise.resolve` which was suggested for ease of reading. I tried a few tweaks here, and this format avoids exposing methods that don't have a use outside this specific call and keeps the order of operations we need to enforce